### PR TITLE
[BEAM-319][BEAM-320] Add distributions module responsible of packagin…

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

…g source and binaries distributions.

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @davorbonaci

File: distributions/pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+              <execution>
+                <id>src</id>
+                <phase>package</phase>
+                <goals>
+                  <goal>single</goal>
+                </goals>
+                <configuration>
+                  <finalName>apache-beam-${project.version}</finalName>
+                  <descriptors>
+                    <descriptor>../sdks/java/build-tools/src/main/assembly/src.xml</descriptor>
```
Comment (davorbonaci) [inline distributions/pom.xml]: It now makes sense to move this file over here, outside java-sdk.

**Comment by jbonofre** [comment]
Rebased and updated.

**Comment by davorbonaci** [comment]
LGTM. Merging.

**Pull request merged**